src/polycap-description.c

```diff
@@ -288,12 +252,12 @@ int polycap_description_get_transmission_efficiencies(polycap_description *descr
 		exit(1);
 	}
 	for(i=0; i<n_energies; i++) sum_weights[i] = 0.;
-
-
+printf("Here1\n");
+printf("%f keV - %f keV\n",energies[0],energies[n_energies-1]);
 //OpenMP loop
 #pragma omp parallel \
 	default(shared) \
-	private(icount, i, j) \
+	private(i, j) \
 	firstprivate(description, n_energies, energies) \
```

Comment:
Ben wat aan het debuggen, maar ik stuit op een probleem dat ik niet direct degelijk weet op te lossen.
Probleem ligt in OpenMP behandelen van de energies pointer. Voor de parallel loop kan ik perfect de inhoud van energies uitlezen (lijn 256 print).
Echter eens in de parallel loop, krijg ik een Segmentation fault (core dumped).
Nu, naar ik het begrijp zou, doordat energies (first)private is, elke thread zijn eigen unieke pointer naar de energies_values moeten hebben, maar dus zou voor elke thread bvb energies[0] wel gelijk moeten zijn aan elkaar. Ksnap dus niet goed waarom het bij mij lijkt alsof de inhoud waar energies naar wijst verdwenen is.
Heb jij een idee?

Reply:
Khad hetzelfde probleem met icount overigens. Al is dat niet eens een pointer. Toen het als private stond was de waarde ervan binnen de parallel region plots quatsch (ad random -27907687 of 690976 etc), terwijl nu als shared is er geen probleem.

Reply:
Ik zie geen reden om `energies` (first)private te maken eerlijk gezegd. Dat is toch een read-only array in deze functie, dus is er geen probleem als verschillende threads daar simultaan zouden zitten in lezen. shared is wat je hier moet gebruiken.

`icount` is effectief een constante (alhoewel je ze niet zo declareert), dus ook read-only, dus ook shared. Variabelen die als private zijn gemarkeerd, zijn thread-specifiek, en worden niet geinitialiseerd, waardoor hun waarde onvoorspelbaar is. firstprivate is zoals private, maar dan met initialisatie met de laatste waarde van de gelijknamige variabele voor de openmp loop betreden wordt.

Reply:
Ja, maar zelfs als ik er niks over zeg, en het dus default(shared) is werkt het niet :(

Reply:
Hmm, zal eens bezien...